File: 코딩 기초 트레이닝/왼쪽 오른쪽-p181890/왼쪽 오른쪽.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181890

def solution(str_list):
    answer = []

    l = -1
    r = -1

    if 'l' in str_list:
        l = str_list.index('l')

    if 'r' in str_list:
        r = str_list.index('r')


    if l > -1 and r > -1:
        if l < r :
            return str_list[0:l]
        else:
            return str_list[r+1:]
    elif l > -1 and r < 0:
        return str_list[0:l]
    elif r > -1 and l < 0:
        return str_list[r+1:]
    else:
        return []

    return answer

File: 코딩 기초 트레이닝/왼쪽 오른쪽-p181890/왼쪽 오른쪽.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181890

function solution(str_list) {
    let lIndex = str_list.indexOf('l');
    let rIndex = str_list.indexOf('r');

    if(lIndex > -1 && rIndex > -1) {
        if(lIndex < rIndex) {
            return str_list.slice(0, lIndex);
        } else {
            return str_list.slice(rIndex+1,);
        }
    } else {
        if(lIndex > -1) {
            return str_list.slice(0, lIndex);
        } else if(rIndex > -1) {
            return str_list.slice(rIndex+1,);
        } else {
            return [];
        }
    }
}
